core/src/main/java/com/oath/oak/Handle.java

```diff
@@ -51,21 +51,25 @@ boolean remove(MemoryManager memoryManager) {
         return true;
     }
 
-    boolean put(V newVal, OakSerializer<V> serializer, MemoryManager memoryManager) {
+    <V> boolean put(V newVal, OakSerializer<V> serializer, MemoryManager memoryManager) {
         writeLock.lock();
         if (isDeleted()) {
             writeLock.unlock();
             return false;
         }
+        innerPut(newVal, serializer, memoryManager);
+        writeLock.unlock();
```

Comment:
I know this was in the code before, but this may be a good time to consider:
`innerPut` calls serializer methods, which we don't necessarily have control over, and potentially shuffles memory, which is also risky. I think we should check for exceptions and unlock if anything bad happens.